packages/machine/test/integration/lifecycle/depositor.ts

```diff
@@ -0,0 +1,232 @@
+import * as cf from "@counterfactual/cf.js";
+import { TestResponseSink } from "../test-response-sink";
+import { ethers } from "ethers";
+ import {
+    UNUSED_FUNDED_ACCOUNT
+  } from "../../utils/environment";
+
+
+/**
+ * A collection of staic methods responsible for "depositing", i.e., running
+ * the intsall protocol with  "balance refund/withdraw" app, and ensuring
+ * the machine state was correctly modified.
+ */
+export class Depositor {
+  public static async makeDeposits(
+    peerA: TestResponseSink,
+    peerB: TestResponseSink
+  ): Promise<any> {
+    await Depositor.deposit(
+      peerA,
+      peerB,
+      ethers.utils.bigNumberify(10),
+      ethers.utils.bigNumberify(0)
+    );
+    await Depositor.deposit(
+      peerB,
+      peerA,
+      ethers.utils.bigNumberify(5),
+      ethers.utils.bigNumberify(10)
+    );
+  }
+
+  /**
+   * @param amountA is the amount wallet A wants to deposit into the channel.
+   * @param amountBCumualtive is the amount wallet B already has in the channel,
+   *        i.e., the threshold for the balance refund.
+   */
+  public static async deposit(
+    peerA: TestResponseSink,
+    peerB: TestResponseSink,
+    amountA: ethers.utils.BigNumber,
+    amountBCumlative: ethers.utils.BigNumber
+  ) {
+    const cfAddr = await Depositor.installBalanceRefund(
+      peerA,
+      peerB,
+      amountBCumlative
+    );
+    await Depositor.uninstallBalanceRefund(
+      cfAddr,
+      peerA,
+      peerB,
+      amountA,
+      amountBCumlative
+    );
+  }
+
+  public static async installBalanceRefund(
+    peerA: TestResponseSink,
+    peerB: TestResponseSink,
+    threshold: ethers.utils.BigNumber
+  ) {
+    const msg = Depositor.startInstallBalanceRefundMsg(
+      peerA.signingKey.address!,
+      peerB.signingKey.address!,
+      threshold
+    );
+    const response = await peerA.runProtocol(msg);
+    expect(response.status).toEqual(cf.legacy.node.ResponseStatus.COMPLETED);
```

Comment:
Seems odd to me that the `ResponseStatus` type is in `cf.legacy.node`.

Reply:
`ResponseStatus` is in a few places elsewhere as well - we should fix those in a separate PR